kernel/work.c

```diff
@@ -597,6 +597,8 @@ static void work_queue_main(void *workq_ptr, void *p2, void *p3)
 			work = CONTAINER_OF(node, struct k_work, node);
 			flag_set(&work->flags, K_WORK_RUNNING_BIT);
 			flag_clear(&work->flags, K_WORK_QUEUED_BIT);
+			/* parasoft suppress item MISRAC2012-DIR_4_1-f */
+			/* Check for details in Github #35664 */
```

Comment:
I think it's better to describe the problem other than referencing Github in code.

Reply:
So, I can put explanation of the @de-nordic into the code comment. Because he explained it very clear, the reason why it is false-positive.

Reply:
@gmarull Sure, I will add a brief comment there. I think for more details I can add commit number instead of PR number on Github.